cartographer_ros/cartographer_ros/node_main.cc

```diff
@@ -63,104 +63,104 @@ void Run() {
-    laser_scan_subscriber = node.node_handle()->subscribe(
-        kLaserScanTopic, kInfiniteSubscriberQueueSize,
-        boost::function<void(const sensor_msgs::LaserScan::ConstPtr&)>(
-            [&](const sensor_msgs::LaserScan::ConstPtr& msg) {
+    laser_scan_subscriber = node.node_handle()->create_subscription<sensor_msgs::msg::LaserScan>(
+        kLaserScanTopic,
+        [&](sensor_msgs::msg::LaserScan::ConstSharedPtr msg) {
               node.map_builder_bridge()
                   ->sensor_bridge(trajectory_id)
                   ->HandleLaserScanMessage(kLaserScanTopic, msg);
-            }));
+        }, rmw_qos_profile_default);
     expected_sensor_ids.insert(kLaserScanTopic);
   }
+
+  rclcpp::Subscription<sensor_msgs::msg::MultiEchoLaserScan>::SharedPtr multi_echo_laser_scan_subscriber;
   if (options.use_multi_echo_laser_scan) {
-    laser_scan_subscriber = node.node_handle()->subscribe(
-        kMultiEchoLaserScanTopic, kInfiniteSubscriberQueueSize,
-        boost::function<void(const sensor_msgs::MultiEchoLaserScan::ConstPtr&)>(
-            [&](const sensor_msgs::MultiEchoLaserScan::ConstPtr& msg) {
+    multi_echo_laser_scan_subscriber = node.node_handle()->create_subscription<sensor_msgs::msg::MultiEchoLaserScan>(
+        kMultiEchoLaserScanTopic,
+        [&](sensor_msgs::msg::MultiEchoLaserScan::ConstSharedPtr msg) {
               node.map_builder_bridge()
                   ->sensor_bridge(trajectory_id)
-                  ->HandleMultiEchoLaserScanMessage(kMultiEchoLaserScanTopic,
-                                                    msg);
-            }));
+                  ->HandleMultiEchoLaserScanMessage(kMultiEchoLaserScanTopic, msg);
+        }, rmw_qos_profile_default);
     expected_sensor_ids.insert(kMultiEchoLaserScanTopic);
   }
 
+  rmw_qos_profile_t custom_qos_profile = rmw_qos_profile_default;
+
+  custom_qos_profile.depth = 1;
+  custom_qos_profile.reliability = RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT;
+  custom_qos_profile.history = RMW_QOS_POLICY_HISTORY_KEEP_LAST;
+
   // For 3D SLAM, subscribe to all point clouds topics.
-  std::vector<::ros::Subscriber> point_cloud_subscribers;
+  std::vector<rclcpp::Subscription<sensor_msgs::msg::PointCloud2>::SharedPtr> point_cloud_subscribers;
   if (options.num_point_clouds > 0) {
     for (int i = 0; i < options.num_point_clouds; ++i) {
-      string topic = kPointCloud2Topic;
-      if (options.num_point_clouds > 1) {
-        topic += "_" + std::to_string(i + 1);
-      }
-      point_cloud_subscribers.push_back(node.node_handle()->subscribe(
-          topic, kInfiniteSubscriberQueueSize,
-          boost::function<void(const sensor_msgs::PointCloud2::ConstPtr&)>(
-              [&, topic](const sensor_msgs::PointCloud2::ConstPtr& msg) {
+      //string topic = kPointCloud2Topic;
+      //if (options.num_point_clouds > 1) {
+      //  topic += "_" + std::to_string(i + 1);
+      //}
+      //point_cloud_topics.push_back(topic);
```

Comment:
Should this be removed, or uncommented in the future? If the latter, please add a TODO or at least a comment indicating why it is still here and when it could be restored.